### PR TITLE
Add claiming verification to ClaimRays and squash bugs

### LINKED ISSUE #16808: Add claiming verification to ClaimRays
## Reason
Currently, if claiming fails within ClaimRays (rays aren't claimed or rays are claimed multiple times), a useful error isn't produced. You just end up with an awful crash.

## Design
Verify that the claiming succeeded (all rays were claimed once and only once) within ClaimRays.

## Impact
Produce more useful errors and avoid a hard crash.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #16808

### CODE REVIEW

File: modules/ray_tracing/src/raytracing/ClaimRays.C [hunk 1]
```diff
@@ -209,3 +219,92 @@ ClaimRays::buildPointNeighbors()
...
+    }
+  };
+
+  // Build the local_map
+  add_to_local_map(_rays, false);
+  add_to_local_map(_local_rays, true);
+
+  // Build the structure to send the local generation/claiming information to rank 0
+  std::map<processor_id_type, std::vector<std::pair<RayID, char>>> send_info;
+  send_info[0] = std::vector<std::pair<RayID, char>>(local_map.begin(), local_map.end());
```
Comment (aeslaughter) [inline modules/ray_tracing/src/raytracing/ClaimRays.C:270]: This should work...
```suggestion
  send_info[0].emplace(local_map.begin(), local_map.end());
```

**Review by aeslaughter — DISMISSED** [review]

File: modules/ray_tracing/src/raytracing/ClaimRays.C (hunk 1 above)
Comment (loganharbour) [reply modules/ray_tracing/src/raytracing/ClaimRays.C:270]: Your comment disappeared? I believe you mean this (it's a map):
```
  send_info.emplace(std::piecewise_construct,
                    std::forward_as_tuple(0),
                    std::forward_as_tuple(local_map.begin(), local_map.end()));
```

Comment (loganharbour) [reply modules/ray_tracing/src/raytracing/ClaimRays.C]: Done and done 👍🏼

File: modules/ray_tracing/src/raytracing/ClaimRays.C [hunk 2]
```diff
@@ -209,3 +219,94 @@ ClaimRays::buildPointNeighbors()
...
+
+  // Rank 0 will make sure everything looks good
+  if (_pid == 0)
+    for (const auto & id_pairs_pair : global_map)
+    {
+      const RayID id = id_pairs_pair.first;
+      const std::vector<std::pair<processor_id_type, char>> & pid_claimed_pairs =
+          id_pairs_pair.second;
+
+      std::vector<processor_id_type> claimed_pids;
```
Comment (GiudGiud) [inline modules/ray_tracing/src/raytracing/ClaimRays.C:296]: the pids arent claimed, they claim. Having trouble finding a better name though

File: modules/ray_tracing/src/raytracing/ClaimRays.C (hunk 1 above)
Comment (aeslaughter) [reply modules/ray_tracing/src/raytracing/ClaimRays.C]: I realized by code suggestion was just wrong.

**Review by aeslaughter — DISMISSED** [review]

**Review by GiudGiud — DISMISSED** [review]
Was _inflated_neighbor_bboxes just unused? or did you use it in Mock

File: modules/ray_tracing/src/raytracing/ClaimRays.C (hunk 2 above)
Comment (loganharbour) [reply modules/ray_tracing/src/raytracing/ClaimRays.C:296]: 🤷🏼‍♂️ close enough

**Review by GiudGiud — APPROVED** [review]

**Comment by loganharbour** [comment]
> Was _inflated_neighbor_bboxes just unused? or did you use it in Mock

Yeah. Just in MOC. It's been moved to there now

**Pull request merged**